kong/dao/db/cassandra.lua

```diff
@@ -357,7 +357,7 @@ local function check_unique_constraints(self, table_name, constraints, values, p
 
   for col, constraint in pairs(constraints.unique) do
     -- Only check constraints if value is non-null
-    if values[col] ~= nil then
+    if values[col] ~= nil and values[col] ~= ngx.null then
```

Comment:
I thought it was the `schema_validator`'s responsibility to nullify the column value if `ngx.null` (JSON NULL) is given? As per the existing implementation [here](https://github.com/Kong/kong/blob/master/kong/dao/schemas_validation.lua#L102-L104) and [here](https://github.com/Kong/kong/blob/master/kong/dao/schemas_validation.lua#L120-L122)?

Reply:
Well, here we need to pass something to DAO. If we `nil` then that field will not be updated (or `nullified`). If we set it to `cassandra.null` then the generic implementations needs to know about the database. `ngx.null` is common way to tell the strategies to do something. Serialize function already handles this: https://github.com/Kong/kong/blob/12244ec4e9bb3587a9e34608ff83c9deaf4a8ec3/kong/dao/db/cassandra.lua#L314

Reply:
Ah true. Alright, then I think all we need would be a regression test for this?

Reply:
That's right. I will be adding it.

Reply:
@thibaultcha the test was added.